compiler/fir/resolve/src/org/jetbrains/kotlin/fir/resolve/transformers/FirCallCompletionResultsWriterTransformer.kt

```diff
@@ -90,11 +90,13 @@ class FirCallCompletionResultsWriterTransformer(
             // e.g. `T::toString` where T is a generic type.
             // in these cases we should report an error on
             // the calleeReference.source which is not a fake source.
-            // uncommenting `?.fakeElement...` here removes reports
-            // of OTHER_ERROR from tests.
             buildErrorTypeRef {
-                source = calleeReference.source //?.fakeElement(FirFakeSourceElementKind.ImplicitTypeRef)
-                diagnostic = ConeSimpleDiagnostic("Callee reference to candidate without return type: ${declaration.render()}")
+                source = calleeReference.source?.fakeElement(FirFakeSourceElementKind.ImplicitTypeRef)
+                diagnostic =
+                    if (declaration is FirTypeParameter)
+                        ConeTypeParameterInQualifiedAccess(declaration.symbol)
```

Comment:
It looks like reified type parameters are not reported. For example

```
inline fun <reified T> foo() {
    T::toString // error
}
```

Reply:
Thanks, updated `compiler/testData/diagnostics/tests/generics/tpAsReified/InFunction.kt` as it contains similar (yet more undetected) cases.